daemon/snap.go

```diff
@@ -168,27 +170,131 @@ type screenshotJSON struct {
 	Height int64  `json:"height,omitempty"`
 }
 
-func mapLocal(about aboutSnap) map[string]interface{} {
-	localSnap, snapst := about.info, about.snapst
-	status := "installed"
-	if snapst.Active && localSnap.Revision == snapst.Current {
-		status = "active"
+type bySnapApp []*snap.AppInfo
+
+func (a bySnapApp) Len() int      { return len(a) }
+func (a bySnapApp) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a bySnapApp) Less(i, j int) bool {
+	iName := a[i].Snap.Name()
+	jName := a[j].Snap.Name()
+	if iName == jName {
+		return a[i].Name < a[j].Name
+	}
+	return iName < jName
+}
+
+// this differs from snap.SplitSnapApp in the handling of the
+// snap-only case:
+//   snap.SplitSnapApp("foo") is ("foo", "foo"),
+//   splitAppName("foo") is ("foo", "").
+func splitAppName(s string) (snap, app string) {
```

Comment:
you could have still use that former and just checked and transformed the result instead of having direct logic, I think I would prefer the reuse (better isolation of knowledge, but it's a nitpick)